osf_tests/test_analytics.py

```diff
@@ -42,3 +44,125 @@ def test_increment_user_activity_counters(self):
+    page_counter, created = PageCounter.objects.select_for_update().get_or_create(_id=page_counter_id)
+    page_counter.date = {u'2018/02/04': {u'total': 41, u'unique': 33}}
+    page_counter.save()
+    return page_counter
+
+@pytest.fixture()
+def page_counter2(project, file_node2):
+    page_counter_id = 'download:{}:{}'.format(project._id, file_node2.id)
+    page_counter, created = PageCounter.objects.select_for_update().get_or_create(_id=page_counter_id)
+    page_counter.date = {u'2018/02/04': {u'total': 4, u'unique': 26}}
+    page_counter.save()
+    return page_counter
+
+@pytest.fixture()
+def page_counter_for_individual_version(project, file_node3):
+    page_counter_id = 'download:{}:{}:0'.format(project._id, file_node3.id)
+    page_counter, created = PageCounter.objects.select_for_update().get_or_create(_id=page_counter_id)
+    page_counter.date = {u'2018/02/04': {u'total': 1, u'unique': 1}}
+    page_counter.save()
+    return page_counter
+
+
+class TestPageCounter:
+
+    @mock.patch('osf.models.analytics.session')
+    def test_download_update_counter(self, mock_session, project, file_node):
+        mock_session.data = {}
+        page_counter_id = 'download:{}:{}'.format(project._id, file_node.id)
+
+        PageCounter.update_counter(page_counter_id, {})
+
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 1
+        assert page_counter.unique == 1
+
+        PageCounter.update_counter(page_counter_id, {})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 2
+        assert page_counter.unique == 1
+
+    @mock.patch('osf.models.analytics.session')
+    def test_download_update_counter_contributor(self, mock_session, user, project_with_contrib, file_node):
+        mock_session.data = {'auth_user_id': user._id}
+        page_counter_id = 'download:{}:{}'.format(project_with_contrib._id, file_node.id)
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project_with_contrib.contributors})
+
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project_with_contrib.contributors})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+    def test_get_all_downloads_on_date(self, page_counter, page_counter2):
+
+        total_downloads = PageCounter.get_all_downloads_on_date('2018/02/04')
+
+        assert total_downloads == 45
+
+    def test_get_all_downloads_on_date_exclude_versions(self, page_counter, page_counter2, page_counter_for_individual_version):
+        total_downloads = PageCounter.get_all_downloads_on_date('2018/02/04')
+
+        assert total_downloads == 45
+
+    def test_download_all_versions_regex(self):
```

Comment:
This test doesn't actually need the database. So we can save a few seconds of database setup/teardown by explicitly marking the classes in this file that need a database and moving this one to its own class that doesn't get marked

```python
@pytest.mark.django_db
class TestAnalytics(..)

@pytest.mark.django_db
class TestPageCounter

class TestPageCounterRegex:
```